api/v1beta2/network_types.go

```diff
@@ -510,6 +510,18 @@ func (s Subnets) FilterPrivate() (res Subnets) {
 	return
 }
 
+// FilterPrimary returns a slice containing all subnets that do not have the
+// sigs.k8s.io/cluster-api-provider-aws/association: secondary tag. These
+// subnets are intended for the CNI.
+func (s Subnets) FilterPrimary() (res Subnets) {
```

Comment:
I was a bit confused by reading the comment and function name. I’d call it `FilterNonCni`, since your goal is to _exclude_ the CNI subnets from being used for EKS node groups.

```suggestion
// FilterNonCni returns the subnets that are NOT intended for usage with the CNI pod network
// (i.e. do NOT have the `sigs.k8s.io/cluster-api-provider-aws/association=secondary` tag).
func (s Subnets) FilterNonCni() (res Subnets) {
```